crates/vm2/src/instruction_handlers/binop.rs

```diff
@@ -104,7 +108,7 @@ impl Binop for Xor {
 impl Binop for ShiftLeft {
     #[inline(always)]
     fn perform(a: &U256, b: &U256) -> (U256, (), Flags) {
-        let result = *a << b.low_u32() as u8;
+        let result = *a << (b.low_u32() & 0xff);
```

Comment:
```suggestion
        let result = *a << (b.low_u32() % 256);
```
Maybe this would make it clearer that the mask isn't arbitrary.